packages/cms/pages/resources/create/create.tsx

```diff
@@ -195,20 +191,37 @@ export const CreateResource: React.FunctionComponent = () => {
     return <p>Loading ...</p> // show full page loader here.
   }
 
-  // const [errors, setErrors] = useState<AbstractData>({})
-  // const [saving, setSaving] = useState(false)
-  // const [isEditing, setIsEditing] = useState(false)
-  // const [form, setForm] = useState<AbstractData>({})
-  // const [booted, setBooted] = useState(false)
+  const publishResource = async () => {
+    setSaving(true)
+
+    const [response, error] = await createResource(resource, form)
+
+    if (error) {
+      let formErrors: any = {}
+      error?.response?.data?.errors?.forEach(
+        (error: { message: string; field: string }) => {
+          formErrors[error.field] = [error?.message]
+        }
+      )
+
+      setErrors(formErrors)
+      setSaving(false)
+      return
+    }
 
-  console.log('Components', JSON.stringify(window.Tensei.components))
+    setSaving(false)
 
-  const TextComponent = window.Tensei.components.form.Text
+    toast(
+      'Created',
+      <p>{resource?.name.toLowerCase()} have been created successfully</p>
+    )
 
-  // const Component =
-  // window.Tensei.components.form[field.component.form] ||
-  // window.Tensei.components.form.Text
-  // return <Component />
+    setTimeout(() => {
```

Comment:
Why are we waiting 2 seconds?
We shouldn't be redirecting like this. We should use `history.push()` to perform a client-side redirect.

Reply:
The wait is to display the toast message

Reply:
Ohh, yeah ... we only use this when there is need for refreshing data from the api